ranges/max-satisfying.js

```diff
@@ -1,10 +1,12 @@
 const SemVer = require('../classes/semver')
 const Range = require('../classes/range')
+const parseOptions = require('../internal/parse-options')
 
 const maxSatisfying = (versions, range, options) => {
   let max = null
   let maxSV = null
   let rangeObj = null
+  options = parseOptions(options)
```

Comment:
reassigning arguments is a deopt in older v8 engines, and potentially in other non-v8 engines as well; can we make a new variable instead?

Reply:
This deopt is true only for V8 engines, and we need to manipulate `arguments` in the method to cause the deopt, [ref](https://github.com/NathanaelA/v8-Natives/blob/master/opti-killers.md#3-managing-arguments).